crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -1883,9 +1883,76 @@ impl TwoQubitBasisDecomposer {
     }
 }
 
+static MAGIC: [[Complex64; 4]; 4] = [
+    [
+        c64(FRAC_1_SQRT_2, 0.),
+        C_ZERO,
+        C_ZERO,
+        c64(0., FRAC_1_SQRT_2),
+    ],
+    [
+        C_ZERO,
+        c64(0., FRAC_1_SQRT_2),
+        c64(FRAC_1_SQRT_2, 0.),
+        C_ZERO,
+    ],
+    [
+        C_ZERO,
+        c64(0., FRAC_1_SQRT_2),
+        c64(-FRAC_1_SQRT_2, 0.),
+        C_ZERO,
+    ],
+    [
+        c64(FRAC_1_SQRT_2, 0.),
+        C_ZERO,
+        C_ZERO,
+        c64(0., -FRAC_1_SQRT_2),
+    ],
+];
+
+#[pyfunction]
+pub fn two_qubit_local_invariants(unitary: PyReadonlyArray2<Complex64>) -> [f64; 3] {
+    let mat = unitary.as_array();
+    // Transform to bell basis
+    let bell_basis_unitary = transpose_conjugate(aview2(&MAGIC)).dot(&mat.dot(&aview2(&MAGIC)));
+    // Get determinate since +- one is allowed.
+    let det_bell_basis = bell_basis_unitary
+        .view()
+        .into_faer_complex()
+        .determinant()
+        .to_num_complex();
+    let m = bell_basis_unitary.t().dot(&bell_basis_unitary);
+    let mut m_tr2 = m.diag().sum();
+    m_tr2 *= m_tr2;
+    // Table II of Ref. 1 or Eq. 28 of Ref. 2.
+    let g1 = m_tr2 / (16. * det_bell_basis);
+    let g2 = (m_tr2 - m.dot(&m).diag().sum()) / (4. * det_bell_basis);
+
+    // Here we split the real and imag pieces of G1 into two so as
+    // to better equate to the Weyl chamber coordinates (c0,c1,c2)
+    // and explore the parameter space.
+    // Also do a FP trick -0.0 + 0.0 = 0.0
+    [g1.re + 0., g1.im + 0., g2.re + 0.]
+}
+
+#[pyfunction]
+pub fn local_equivalence(weyl: PyReadonlyArray1<f64>) -> PyResult<[f64; 3]> {
+    let weyl = weyl.as_slice()?;
```

Comment:
This can fail if the input array isn't contiguous, I think.

Reply:
Yeah, that's why it returns a `PyResult`. As it's internal I didn't think it mattered, although I debated moving to just passing in 3 floats in a `[f64; 3]` directly because there are only 3 values. I can make that change if you think it's clearer.

Reply:
Sorry, I guess I didn't make the thought clear: this returns a Python exception, except it's needless because we don't _need_ this to be contiguous.  Even if we keep passing an array, we can still do what we need with strided access, right?

Reply:
Ah ok, yeah I can use an `ArrayView1` here instead too. Using arrays here just felt like overkill here for something that is just 3 floats

Reply:
Used `as_array()` here to keep everything as an arrayview1: https://github.com/Qiskit/qiskit/pull/12739/commits/21279873e74df243776b390de8bc548594576607